iot/manager/manager_test.go

```diff
@@ -52,54 +68,149 @@ func setup(m *testing.M) {
 	if err != nil {
-		log.Fatalf("Could not create topic: %v", err)
+		log.Fatalf("Could not create registry: %v\n", err)
 	}
-	fmt.Printf("Topic created: %v\n", t)
 }
 
 func shutdown() {
 	ctx := context.Background()
 
 	t := client.Topic(topicID)
 	if err := t.Delete(ctx); err != nil {
-		log.Fatalf("Could not delete topic: %v", err)
+		log.Fatalf("Could not delete topic: %v\n", err)
 	}
 	fmt.Printf("Deleted topic: %v\n", t)
+
+	if _, err := deleteRegistry(os.Stdout, projectID, region, registryID); err != nil {
+		log.Fatalf("Could not delete registry: %v\n", err)
+	}
 }
 
-func TestSendCommand(t *testing.T) {
-	// Generate UUID v1 for test registry and device
-	registryUUID, _ := uuid.NewRandom()
-	deviceUUID, _ := uuid.NewRandom()
+func TestCreateRegistry(t *testing.T) {
+	testRegistryID := createIDForTest("registry")
+
+	testutil.Retry(t, 10, 10*time.Second, func(r *testutil.R) {
+		buf := new(bytes.Buffer)
+		registry, err := createRegistry(buf, projectID, region, testRegistryID, topicName)
+		if err != nil {
+			r.Errorf("Could not create registry: %v\n", err)
+
+		} else {
+			if registry.Id != testRegistryID {
+				r.Errorf("Created registry, but registryID is wrong. Got %q, want %q", registry.Id, testRegistryID)
+			}
 
-	region := "us-central1"
-	registryID := "golang-test-registry-" + registryUUID.String()
-	deviceID := "golang-test-device-" + deviceUUID.String()
+			got := buf.String()
+			want := fmt.Sprintf("Created registry:\n\tID: %s", testRegistryID)
+			if !strings.Contains(got, want) {
+				r.Errorf("CreateRegistry got %s, want substring %q", got, want)
+			}
 
-	topic := client.Topic(topicID)
+			deleteRegistry(buf, projectID, region, testRegistryID)
+		}
+	})
+}
 
-	var buf bytes.Buffer
+func TestGetRegistry(t *testing.T) {
+	testRegistryID := createIDForTest("registry")
 
-	if _, err := createRegistry(&buf, projectID, region, registryID, topic.String()); err != nil {
-		t.Fatalf("Could not create registry: %v", err)
-	}
+	testutil.Retry(t, 10, 10*time.Second, func(r *testutil.R) {
+		buf := new(bytes.Buffer)
+		_, err := createRegistry(buf, projectID, region, testRegistryID, topicName)
+		if err != nil {
+			r.Errorf("Could not create registry: %v\n", err)
+		}
+
+		_, err = getRegistry(buf, projectID, region, testRegistryID)
+		if err != nil {
+			r.Errorf("Could not get registry: %v\n", err)
+		}
+
+		got := buf.String()
+		want := "Got registry:\n\tID: " + testRegistryID
+		if !strings.Contains(got, want) {
+			r.Errorf("GetRegistry got %s, want substring %q", got, want)
+		}
+
+		deleteRegistry(buf, projectID, region, testRegistryID)
+	})
+}
+
+func TestListRegistries(t *testing.T) {
+	testRegistryID := createIDForTest("registry")
 
-	if _, err := createUnauth(&buf, projectID, region, registryID, deviceID); err != nil {
+	testutil.Retry(t, 10, 10*time.Second, func(r *testutil.R) {
+		buf := new(bytes.Buffer)
+		_, err := createRegistry(buf, projectID, region, testRegistryID, topicName)
+		if err != nil {
```

Comment:
Prefer inline declaration of errors. It avoids accidental shadowing of errors.

```suggestion
		if _, err := createRegistry(buf, projectID, region, testRegistryID, topicName); err != nil {
```